packages/block-editor/src/components/block-navigation/use-block-navigation-drop-zone.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * WordPress dependencies
+ */
+import { __unstableUseDropZone as useDropZone } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { getDistanceToNearestEdge } from '@wordpress/dom';
+import { useEffect, useRef, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import useOnHTMLDrop from '../use-block-drop-zone/use-on-html-drop';
+import useOnFileDrop from '../use-block-drop-zone/use-on-file-drop';
+import useOnBlockDrop from '../use-block-drop-zone/use-on-block-drop';
```

Comment:
Not sure why I made these into three hooks. I think it makes sense to combine them into one that returns three callbacks.

Reply:
Split out into https://github.com/WordPress/gutenberg/pull/24649.